dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -516,6 +573,48 @@ def _perform_t_test(
         results["welch"]["p-value"] = float(welch_p_val)
         return results
 
+    @staticmethod
+    def _calculate_psi(
+        mean1: float,
+        sum1: float,
+        histogram1: np.ndarray,
+        mean2: float,
+        sum2: float,
+        histogram2: np.ndarray,
+    ) -> Optional[float]:
+        self_count = sum1 / mean1
+        other_count = sum2 / mean2
+
+        psi_value = 0
+        invalid_stats = False
+
+        if isinstance(histogram1["bin_counts"], type(None)) or isinstance(
+            histogram1["bin_edges"], type(None)
+        ):
+            warnings.warn(
+                "No data available in the histograms for calculating `PSI`",
+                RuntimeWarning,
+            )
+            invalid_stats = True
+
+        if invalid_stats:
+            return None
+
+        for iter_value, bin_count in enumerate(histogram1["bin_counts"]):
+
+            self_percent = bin_count / self_count
+            other_percent = histogram2["bin_counts"][iter_value] / other_count
+            if (self_percent == other_percent) and self_percent == 0:
+                continue
+
+            iter_psi = (other_percent - self_percent) * np.log(
+                other_percent / self_percent
+            )
+            if iter_psi and iter_psi != float("inf"):
+                psi_value += iter_psi
+
+        return psi_value
+
```

Comment:
`staticmethod` for calculating the `psi` value between two column profile objects